compiler/qsc_frontend/src/resolve.rs

```diff
@@ -667,10 +667,22 @@ impl AstVisitor<'_> for With<'_> {
             }
             ast::StmtKind::Qubit(_, pat, init, block) => {
                 ast_visit::walk_qubit_init(self, init);
-                // The binding is valid after end of the statement.
-                self.resolver.bind_pat(pat, stmt.span.hi);
                 if let Some(block) = block {
-                    self.visit_block(block);
+                    self.with_scope(block.span, ScopeKind::Block, |visitor| {
+                        // The binding is valid after end of the statement, but scoped to the current block.
+                        visitor.resolver.bind_pat(pat, stmt.span.hi);
```

Comment:
`block.span.lo` would be the more correct offset to use here (but if you take my suggestion to use `with_pat`, you don't need to worry about it)

Reply:
I'll use your other suggestion.